polaris-react/src/styles/foundation/_focus-ring.scss

```diff
@@ -1,55 +1,30 @@
 /// Sets the focus ring for an interactive element
-/// @param {String} $size - The size of the border radius on the focus ring.
 /// @param {String} $style - Focus ring state.
+///
+/// @deprecated No longer needed
 /// @param {Number} $border-width - Optional negative offset (can be used to ensure the outline wraps the border).
-/// @param {Number} $delay - The transition delay in milliseconds.
 ///
+/// @deprecated No longer needed
+/// @param {String} $size - The size of the border radius on the focus ring.
 
-@mixin focus-ring($size: 'base', $border-width: 0, $style: 'base') {
+@mixin focus-ring($style: 'base', $size: 'base', $border-width: 0) {
   $stroke: 2px;
-  // calc does not like performing addition with a unitless number (`0`, NOT `0px`) This makes sure that we can handle unitless zeros by not trying to do math with them.
-  $offset: if($border-width == 0, 1px, calc(#{$border-width} + 1px));
-  $border-radius: if(
-    $size == 'wide',
-    var(--p-border-radius-2),
-    var(--p-border-radius-1)
-  );
-  $negative-offset: calc(-1 * #{$offset});
+  outline: $stroke solid transparent;
+  transition: outline var(--p-motion-duration-100) var(--p-motion-ease);
 
-  @if $style == 'base' {
-    position: relative;
-
-    &::after {
-      content: '';
-      position: absolute;
-      // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
-      z-index: 1;
-      top: $negative-offset;
-      right: $negative-offset;
-      bottom: $negative-offset;
-      left: $negative-offset;
-      display: block;
-      pointer-events: none;
-      box-shadow: 0
-        0
-        0
-        $negative-offset
-        var(--p-color-border-interactive-focus);
-      transition: box-shadow var(--p-motion-duration-100) var(--p-motion-ease);
-      border-radius: $border-radius;
-      @content;
-    }
-  } @else if $style == 'focused' {
-    &::after {
-      box-shadow: 0 0 0 $stroke var(--p-color-border-interactive-focus);
-      outline: var(--p-border-width-1) solid transparent;
-      @content;
```

Comment:
I think this might still be necessary. Compared functionality in Combobox (which renders Listbox) and looks like when you use the arrow key to go through listbox options, the focus ring flash still occurs.

Prod Storybook:
![combobox-prod](https://github.com/Shopify/polaris/assets/26749317/ef0a0e4b-52a9-454c-8835-679338138f86)

PR Storybook:
![combobox-pr](https://github.com/Shopify/polaris/assets/26749317/69166158-9e31-4230-b933-72e35f96afe9)

Reply:
That seems to be a separate issue. There aren't any implementations of using `focus-ring with {@content}` overrides in our repo. I think we need a better way to disable focus ring on the TextOption component. Looking into it now